repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -145,4 +151,71 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
   def getSplittableHeaderScala(headerName: String): List[String] = getHeadersScala(headerName).foldLeft(List.empty[String])((list, s) => list ++ s.split(","))
 
   override def getSplittableHeaders(headerName: String): util.List[String] = getSplittableHeaderScala(headerName).asJava
+
+  /**
+    * @return a [[StringBuffer]] containing the reconstructed URL for this request (note that
+    *         mutation of this [[StringBuffer]] will have no effect on the request URL)
+    */
+  override def getRequestURL: StringBuffer = {
+    val url: StringBuffer = new StringBuffer(getScheme).append("://").append(getServerName)
+
+    if (getServerPort > 0 && ((HTTP.equalsIgnoreCase(getScheme) && getServerPort != 80) || (HTTPS.equalsIgnoreCase(getScheme) && getServerPort != 443))) {
+      url.append(':').append(getServerPort)
+    }
+
+    if (Option(getRequestURI).exists(_.nonEmpty)) {
+      url.append(getRequestURI)
+    }
+
+    url
+  }
+
+  override def getRequestURI: String = requestUri
+
+  def setRequestURI(uri: String): Unit = {
```

Comment:
Hm, Jetty and Glassfish both build the URL using the getRequestUri() method. So we can probably get away with not implementing that.